### ISSUE TITLE
Automating new cookie

### ISSUE THREAD

**Issue by vic18t** [alexxsalazar/Nvidia3080_BB_bot#10]
Hi Alex, would you be able to automatically get a new cookie from Best Buy every few hours? Manually getting a new one every few hours is a pain :P It would suck if the bot failed because I didn't refresh a cookie on time either.

**Comment by alexxsalazar** [comment]
Yeah I get it. Maybe I can look into it more some time next week. To be honest I just made it real quick and put it out hoping it would help others make their own version. We’ll see though.

**Comment by vic18t** [comment]
Thank you, appreciate it. Yeah, doesn’t seem like anyone is picking up the torch, but you!

**Comment by Infinitay** [comment]
https://github.com/zedd3v/abck-go
https://github.com/zedd3v/abck

Potential resources for anyone willing to adopt it for python

**Comment by scyco52** [comment]
if i was only smart enough to make this work 1 month non stop and cant get card oh well thanks anyways im sure it works if you know how to make it work. im pretty good with computers too. lol

**Comment by AdamOC3** [comment]
> if i was only smart enough to make this work 1 month non stop and cant get card oh well thanks anyways im sure it works if you know how to make it work. im pretty good with computers too. lol

I think Best Buy has new measures in place.  I had bot going when some cards dropped and they got stuck at "add to cart" action.

**Comment by scyco52** [comment]
I'm sure they do. I've been looking around I cant.get.a.damn card its.crazy and I'm no scalper or I could write my own bot. I just can't find one card.that isnt.gone in a second. I'm sure.webites are trying to pit measures in place but we still don't stand a chance.against bots.Thanks JackSent from my T-Mobile 4G LTE Device

-------- Original message --------

**Comment by alexxsalazar** [comment]
Yeah it got stuck because the cookie values were no longer valid. Whenever the cookie values go bad it does that. That's only annoying part about Best Buy. My friend was able to get one with it this past week so I think the bot itself is still good. Sorry y'all. If I had the time I would work on automating the cookie portion but that'd most likely big time commitment for something that would eventually break. Akamai data isn't fun to generate. Even the akamai programs listed above are broken. I believe they just keep changing the way they validate sessions so I would have to keep adapting just to generate cookies.

**Comment by vic18t** [comment]
Could you create a script that just scrapes new cookies from BB?

**Comment by alexxsalazar** [comment]
No someone would have do the surfing to get appropriate cookies. Akamai takes a bunch of data to determine whether or not you're a bot. Probs like window size, how fast you scroll, where you scroll to etc etc. You could generate the data through a program but doing that would probably take me some time. I would have to look through a bunch of sessions and see what data it took to validate the session. Then find ways to generate each of the values necessary. If I was able to make it, it would just end up being used by scalpers until it breaks anyways. Not only for Best Buy but also other sites that require Akamai data like shoe sites for example. I could, in theory, just give out a compiled version but then that would be sketchy and it wouldn't be very hard to reverse engineer anyways. Just doesn't seem like a good idea

**Comment by vic18t** [comment]
No worries. The cookies don’t expire for at least 4 hours that I have tested. Thanks for the support!

**Comment by alexxsalazar** [comment]
Yeah no worries. Sorry it's a pain haha feel bad for not doing more

**Comment by AdamOC3** [comment]
> Yeah it got stuck because the cookie values were no longer valid. Whenever the cookie values go bad it does that. That's only annoying part about Best Buy. My friend was able to get one with it this past week so I think the bot itself is still good. Sorry y'all. If I had the time I would work on automating the cookie portion but that'd most likely big time commitment for something that would eventually break. Akamai data isn't fun to generate. Even the akamai programs listed above are broken. I believe they just keep changing the way they validate sessions so I would have to keep adapting just to generate cookies.

Ok thanks I'll keep trying :)
